### PR TITLE
Deal with deprecation warning

### LINKED ISSUE #4546: Deal with deprecation warning on import
In #4540 I added a check for the matching Theano version to the main `__init__.py`.

Unfortunately we did not notice that the `semver.match` function is deprecated:

```
../../opt/conda/lib/python3.8/site-packages/pymc3/__init__.py:34
  /opt/conda/lib/python3.8/site-packages/pymc3/__init__.py:34: DeprecationWarning: Function 'semver.match' is deprecated. Deprecated since version 2.10.0.  This function will be removed in semver 3. Use the respective 'semver.VersionInfo.match' instead.
    if not semver.match(theano.__version__, ">=1.1.2"):
```

We need to make a follow-up PR and fix that for the `v3` branch.

### PR DESCRIPTION
**Pull request by michaelosthege**
Closes #4546

### CODE REVIEW

**Comment by twiecki** [comment]
```
Run python -m pytest -vv --cov=pymc3 --cov-report=xml --cov-report term --durations=50 $TEST_SUBSET
ImportError while loading conftest '/home/runner/work/pymc3/pymc3/pymc3/tests/conftest.py'.
pymc3/__init__.py:34: in <module>
    if not semver.Version.parse(theano.__version__).match(">=1.1.2"):
E   AttributeError: module 'semver' has no attribute 'Version'
Error: Process completed with exit code 4.
```

**Comment by michaelosthege** [comment]
Sorry about this mess. It looks like they are renaming `VersionInfo` to `Version`, but `Version` was not yet available.
Well, at least they follow semantic versioning.

Opening PRs directly in Github works.. But as you can see it's easy to mess up

**Review by ricardoV94 — DISMISSED** [review]
LGTM. Would a unittest where we monkeypatch theano versions / locations be an overkill?

Edit: Or even install the wrong version of theano

File: pymc3/__init__.py [hunk 1]
```diff
@@ -31,14 +31,29 @@
...
-        + f"\nThe installed Theano(-PyMC) version ({theano.__version__}) does not match the PyMC3 requirements."
-        + "\nFor PyMC3 to work, Theano must be uninstalled and replaced with Theano-PyMC."
-        + "\nSee https://github.com/pymc-devs/pymc3/wiki for installation instructions.\n"
-        + "!" * 60
-    )
+def _check_backend_version():
+    backend_paths = theano.__spec__.submodule_search_locations
+    try:
+        backend_version = theano.__version__
+    except:
```
Comment (ricardoV94) [inline pymc3/__init__.py:38]: Should we print the exception in case that helps figuring out different issues under the same umbrella?

Comment (michaelosthege) [reply pymc3/__init__.py:38]: The exception in this case is just `AttributeError: module 'theano' has no attribute '__version__'` as seen in https://github.com/pymc-devs/pymc3/issues/4510#issuecomment-802721737

I don't think that would help in any way. And the conclusion would still be the same..

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**